File: ios/Classes/SampleRenderWorker.m
//
//  SampleRender.m
//  opengl_texture
//
//  Created by German Saprykin on 22/4/18.
//

#import "SampleRenderWorker.h"
#import <OpenGLES/ES2/gl.h>

@interface SampleRenderWorker() {
    CGFloat _tick;
}
@end
@implementation SampleRenderWorker

- (void)onCreate {
    
}

- (BOOL)onDraw {
    _tick = _tick + M_PI / 60;
    CGFloat green = (sin(_tick) + 1) / 2;
    glClearColor(0, green, 0, 1);
    
    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
    return YES;
}

- (void)onDispose {
    
}

@end

File: android/src/main/java/com/yourcompany/opengltexture/SampleRenderWorker.java
package com.yourcompany.opengltexture;

import android.opengl.GLES20;

class SampleRenderWorker implements OpenGLRenderer.Worker {

    private double _tick = 0;

    @Override
    public void onCreate() {

    }

    @Override
    public boolean onDraw() {
        _tick = _tick + Math.PI / 60;

        float green = (float) ((Math.sin(_tick) + 1) / 2);

        GLES20.glClearColor(0f, green, 0f, 1f);

        GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT | GLES20.GL_DEPTH_BUFFER_BIT);
        return true;
    }

    @Override
    public void onDispose() {

    }
}
